### PR TITLE
precommit: use Sphinx Lint check

### PR DESCRIPTION
**Pull request by Borda**
Just a suggestion: shall we use this hook instead of local and consecutively fix duplicated reference targets?

cc: @hugovk

### CODE REVIEW

**Comment by hugovk** [comment]
I'm not sure why it was originally added as a local tool, but seems to make more sense as a repo.

Any idea why pygments and restructuredtext_lint were `additional_dependencies`? If we don't need them, let's remove the empty `additional_dependencies`.

About the duplicates, they look valid, because https://github.com/pytest-dev/pytest-rerunfailures/blob/master/CHANGES.rst is like:

# 13.0 (2023-11-22)
## Breaking changes
## Features
## Bug fixes
# 12.0 (2023-07-05)
## Breaking changes
## Features
## Bug fixes

**Comment by hugovk** [comment]
> Any idea why pygments and restructuredtext_lint were `additional_dependencies`? 

Ah, because restructuredtext_lint and rstcheck are different tools :)

How similar are they? 

There's also https://github.com/sphinx-contrib/sphinx-lint (which I help maintain).

**Comment by Borda** [comment]
> There's also https://github.com/sphinx-contrib/sphinx-lint (which I help maintain).

Sound like a good alternative =)

**Comment by icemac** [comment]
I think `restructuredtext_lint` just predates `sphinx-lint`.

**Comment by Borda** [comment]
@icemac @hugovk, mind reviewing this PR? :flamingo:

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -31,11 +31,7 @@ repos:
...
+    - repo: https://github.com/sphinx-contrib/sphinx-lint
+      rev: v0.9.1
       hooks:
-          - id: rst
-            name: rst
-            entry: rst-lint --encoding utf-8
-            files: ^(CHANGES.rst|README.rst)$
-            language: python
-            additional_dependencies: [pygments, restructuredtext_lint]
+        - id: sphinx-lint
```
Comment (hugovk) [inline .pre-commit-config.yaml:37]: I recommend this too, it looks for single backticks in RST, which should usually be double backticks for code formatting (MD uses single).
```suggestion
        - id: sphinx-lint
          args: [--enable=default-role]
```

**Comment by hugovk** [comment]
These look like valid errors, please could you fix them?

```
CHANGES.rst:93: default role used (hint: for inline literals, use double backticks) (default-role)
CHANGES.rst:117: default role used (hint: for inline literals, use double backticks) (default-role)
CHANGES.rst:322: default role used (hint: for inline literals, use double backticks) (default-role)
CHANGES.rst:417: default role used (hint: for inline literals, use double backticks) (default-role)
README.rst:48: default role used (hint: for inline literals, use double backticks) (default-role)
```

**Comment by Borda** [comment]
> These look like valid errors, please could you fix them?

Yes, will check it

**Review by hugovk — APPROVED** [review]

**Pull request merged**